src/figuresClasses.ts

```diff
@@ -1,11 +1,71 @@
-export interface Figure {}
+type Shape = 'triangle' | 'circle' | 'rectangle';
+type Color = 'red' | 'green' | 'blue';
+type GetArea = () => number;
 
-export class Triangle implements Figure {}
+const roundArea = (area: number): number => Math.floor(area * 100) / 100;
 
-export class Circle implements Figure {}
+export interface Figure {
+  shape: Shape;
+  color: Color;
+  getArea: GetArea;
+}
+
+export class Triangle implements Figure {
+  shape: Shape = 'triangle';
+
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    if (a + b <= c) {
+      throw new Error();
+    }
+  }
+
+  getArea(): number {
+    const s: number = (1 / 2) * (this.a + this.b + this.c);
```

Comment:
Just to avoid typing `this` each time, let's extract properties at the top level. 

```
  getArea(): number {
    { ... } = this;
```